enhancements/ingress/configurable-dns-loadbalancerservice.md

```diff
@@ -0,0 +1,245 @@
+)
+```
+
+Re-uses existing state `DNSManaged` of the controller to denote if
+the wildcard *DNSRecord* associated with the controller is managed/unmanaged. 
+
+### Implementation Details/Notes/Constraints
+
+Based on `.loadBalancer.dns`, the controller decides whether to ensure creating
+a wildcard *DNSRecord*. If the customer is updating `.loadBalancer.dns` from 
+`"Managed"` to `"Unmanaged"` the previously created *DNSRecord* will be deleted.
+
+The support provided by the installer and this new field aren't entirely
+mutually exclusive, i.e. if customer has opted to disable DNS management cluster
+wide (as documented
+[here](https://github.com/openshift/installer/blob/master/docs/user/aws/install_upi.md#remove-dns-zones)),
+this field is of no value and need not be set since *DNSRecord* created is silently
+reconciled by the DNS controller.
+
+### Risks and Mitigations
+
+The manual creation of the required DNS records will need to be done prior to
+making this change as the canary controller will try to ensure the if the ingress
+domains are reachable, otherwise the `cluster-ingress-operator` gets into a 
+degraded state. This needs to be documented to avoid unnecessary support cases.
+
+### Drawbacks
+
+N/A 
+
+## Design Details
+
+### Open Questions
+
+Currently, `.spec.domain` in the ingresscontroller cannot be correctly
+updated after creation since updating the [domain](https://github.com/openshift/cluster-ingress-operator/blob/972f09b9dbb181ae5c414da2a990b57c60fde9d8/pkg/operator/controller/ingress/controller.go#L342-L355)
+in the ingresscontroller doesn't result in [updating](https://github.com/openshift/cluster-ingress-operator/blob/972f09b9dbb181ae5c414da2a990b57c60fde9d8/pkg/operator/controller/ingress/dns.go#L97)
+the wildcard *DNSRecord*. 
+> 1. Was this a conscious choice or can this logic be updated?
+
+### Test Plan
+
+- Test by updating the default ingresscontroller domain with a manually
+  created DNS record and ensure correct working of test workload.
+- Test to ensure clean up old DNS records when updating `.loadBalancer.dns` to `Unmanaged`.
+
+### Graduation Criteria
+
+N/A.
+
+#### Dev Preview -> Tech Preview
+
+N/A.  This feature will go directly to GA.
+
+#### Tech Preview -> GA
+
+N/A.  This feature will go directly to GA.
+
+#### Removing a deprecated feature
+
+N/A.
+
+### Upgrade / Downgrade Strategy
+
+On upgrades, `.loadBalancer.dns` will default to `Managed` which is consistent with the
+older versions of the ingress operator.
+
+On downgrades, 
+- If downgrading to an unsupported version while having the `.loadBalancer.dns` set to
+  `Unmanaged` could result in failing to create a wildcard *DNSRecord* and thereby
+  resulting in the `cluster-ingress-operator` in a degraded state. To recover
+  DNS management will have to disabled cluster wide ([docs](https://github.com/openshift/installer/blob/master/docs/user/aws/install_upi.md#remove-dns-zones)).
```

Comment:
```suggestion
  DNS management will have to be disabled cluster wide ([docs](https://github.com/openshift/installer/blob/master/docs/user/aws/install_upi.md#remove-dns-zones)).
```
I think we should only suggest to disable DNS management cluster wide if we have only one (unmanaged DNS) ingress controller. In the event of a mix of managed and unmanaged, this might lead to the cluster not reconciling DNS records it is responsible for.
I also had a question concerning the downgrade: Does the degraded state come from the fact that the controller will try to recreate an existing DNS record (for a domain it doesnt have)? Or does it come from the canaries ? 
In the latter case, canaries should still be able to reach this externally managed DNS wildcard record.

Reply:
> I think we should only suggest to disable DNS management cluster wide if we have only one (unmanaged DNS) ingress controller. In the event of a mix of managed and unmanaged, this might lead to the cluster not reconciling DNS records it is responsible for.

Added that as a condition.

> I also had a question concerning the downgrade: Does the degraded state come from the fact that the controller will try to recreate an existing DNS record (for a domain it doesnt have)? Or does it come from the canaries ?
In the latter case, canaries should still be able to reach this externally managed DNS wildcard record.

This depends on which controller has been configured to unmanaged. 
* If the default controller has the flag, will result in canaries not able to reach the domain. If the external domain has been    created, then it errors when trying to create the DNS record. (currently hinges on [open question](https://github.com/openshift/enhancements/blob/f528e26c377c355164080be3a9df11490b591988/enhancements/ingress/configurable-dns-loadbalancerservice.md#open-questions)) 
* If it is not the default controller, canaries are not in the picture, so it only errors when trying to create the DNS record.